tests/test_repository.py

```diff
@@ -763,9 +768,8 @@ class TestParentRepository(unittest.TestCase):
     """A test to verify that when a parent repository is used the correct one is found."""
 
     def setUp(self):
-        self.testDir = os.path.join(ROOT, 'TestParentRepository')
-        if os.path.exists(self.testDir):
-            shutil.rmtree(self.testDir)
+        self.testDir = tempfile.mkdtemp(dir=ROOT, prefix="TestBasics-")
+        self.testDir = os.path.join(self.testDir, 'TestParentRepository')
```

Comment:
this line shouldn't be here? It's preventing files inside `TestBasics-...` (which should be `TestParentRepository-...`) 
from getting removed.